src/main/java/com/bytebreak/animagic/texture/AnimagicTextureRegion.java

```diff
@@ -5,65 +5,47 @@
 import com.badlogic.gdx.math.Vector2;
 
 public class AnimagicTextureRegion extends TextureRegion {
+    public AnimagicTextureData meta = new AnimagicTextureData();
```

Comment:
I think this should be a final variable.  I don't think it is a good idea to let outside code set this var to null;